src/Paket.Core/NugetConvert.fs

```diff
@@ -228,21 +229,35 @@ let createPackageRequirement (packageName, version, restrictions) sources depend
 let createDependenciesFileR (rootDirectory : DirectoryInfo) nugetEnv mode =
     
     let dependenciesFileName = Path.Combine(rootDirectory.FullName, Constants.DependenciesFileName)
-    
-    let allVersions =
+
+    let allVersionsById =
 
         nugetEnv.NugetProjectFiles
         |> Seq.collect (fun (_,c) -> c.Packages)
         |> Seq.groupBy (fun p -> p.Id)
+
+    let allVersionsByName =
+        allVersionsById
+        |> Seq.map (fun (name, packages) -> name, packages |> Seq.map (fun p -> p.Version) |> Seq.distinct)
+        |> Seq.sortBy (fun (name,_) -> name.ToLower())
+    
+    for (name, versions) in allVersionsByName do
+        if Seq.length versions > 1 
+        then traceWarnfn "Package %s is referenced multiple times in different versions: %A. Paket will choose the latest one." 
+                            name    
+                            (versions |> Seq.map string |> Seq.toList)    
+    let allVersions =
+        allVersionsById
         |> Seq.map (fun (name, packages) -> name, packages |> Seq.map (fun p -> p.Version, p.TargetFramework) |> Seq.distinct)
         |> Seq.sortBy (fun (name,_) -> name.ToLower())
     
     for (name, versions) in allVersions do
-        if Seq.length versions > 1 
+        let targets = versions |> Seq.choose(fun (p,target) -> target)
```

Comment:
Could be some kind of distinct. But if you take distinct and you have both target and version conflict, then again it would hide some details. And maybe target framework should be info only, not warning, as it is not so bad as version conflict.